realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -690,17 +690,173 @@ public void onChange() {
             }
         });
 
-        awaitOrThrow(numberOfInvocation);
+        TestHelper.awaitOrFail(numberOfInvocation);
         mainThread.quit();
     }
 
-    private void awaitOrThrow(CountDownLatch latch) {
-        try {
-            if (!latch.await(5, TimeUnit.SECONDS)) {
-                fail("Latch timed out " + latch);
+    public void testAsyncRealmResultsShouldNotBlockBackgroundCommitNotification() throws Throwable {
```

Comment:
I could use some more comments in this test. The title is good enough, but the flow is really hard to follow